Assets/Tests/InputSystem/CoreTests_Editor.cs

```diff
@@ -2890,27 +2890,10 @@ public void Editor_WhenRunUpdatesInEditModeIsEnabled_InputActionsTriggerInEditMo
         Assert.That(action.ReadValue<float>(), Is.EqualTo(0.75f).Within(0.00001f));
     }
 
-    private static void DisableProjectWideActions()
-    {
-#if UNITY_INPUT_SYSTEM_PROJECT_WIDE_ACTIONS
-        // If the system has project-wide input actions they will also trigger enable/disable via
-        // play mode change triggers above. Hence we adjust extra variable to compensate of
-        // state allocated by project-wide actions.
-        if (InputSystem.actions)
-        {
-            Assert.That(InputActionState.s_GlobalState.globalList.length, Is.EqualTo(1));
-            InputSystem.actions.Disable();
-            InputActionState.DestroyAllActionMapStates();
-        }
-#endif
-    }
-
     [Test]
     [Category("Editor")]
     public void Editor_LeavingPlayMode_DestroysAllActionStates()
```

Comment:
It seems this change allows actions to be enabled earlier which doesn't convolute test case so removed changes I did in recent PR and was also able to remove initial disable added previously. I do not yet fully understand how this is possible but its definitely much better if other state is preserved. I suspect its pushed earlier to the static stack so action states are pushed prior to the test.

Reply:
>I suspect its pushed earlier to the static stack so action states are pushed prior to the test.

I also think the Push/Pop of state between tests is the likely culprit.